lib/configs/imports.ts

```diff
@@ -0,0 +1,122 @@
+/*!
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+import type { ConfigOptions } from '../types.d.ts'
+import type { Linter } from 'eslint'
+
+import perfectionist from 'eslint-plugin-perfectionist'
+import {
+	GLOB_FILES_JAVASCRIPT,
+	GLOB_FILES_TYPESCRIPT,
+	GLOB_FILES_VUE,
+} from '../globs.ts'
+
+/**
+ * Generate imports and exports related ESLint rules.
+ *
+ * @param options - Configuration options
+ */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+export function imports(options: ConfigOptions): Linter.Config[] {
+	return [
+		{
+			name: 'nextcloud/imports/setup',
+			plugins: {
+				perfectionist,
+			},
+		},
+		{
+			name: 'nextcloud/imports/rules',
+			files: [
+				...GLOB_FILES_JAVASCRIPT,
+				...GLOB_FILES_TYPESCRIPT,
+				...GLOB_FILES_VUE,
+			],
+			rules: {
+				// Require file extensions
+				'no-restricted-imports': [
+					'error',
+					{
+						patterns: [
+							{
+								regex: '^(\\.*)/(.+/)*[^/.]+$',
+								message: 'Import is missing the file extension.',
+							},
+						],
+					},
+				],
+				// Sorting of imports
+				'sort-imports': 'off',
+				'perfectionist/sort-imports': [
+					'error',
+					{
+						type: 'natural',
+						newlinesBetween: 'never',
+						groups: [
+							// type first
+							'type',
+							{ newlinesBetween: 'always' },
+							// external things
+							[
+								'builtin',
+								'external',
+								'object',
+							],
+							// everything else which is everything internal
+							'unknown',
+							// Vue components
+							[
+								'nextcloudVue',
+								'vue',
+							],
+							'localVue',
+							{ newlinesBetween: 'always' },
+							// import style from 'my.module.css'
+							'style',
+							// side effect only: import 'sideeffect.js'
+							'side-effect',
```

Comment:
To have JS, then CSS, not JS - CSS - JS again

```suggestion
							// side effect only: import 'sideeffect.js'
							'side-effect',
							// import style from 'my.module.css'
							'style',
```

Reply:
done